src/Aspire.Hosting.Azure/BicepGenerationContext.cs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Azure.Provisioning;
+
+namespace Aspire.Hosting.Azure;
+
+/// <summary>
+/// TODO:
+/// </summary>
+public sealed class BicepGenerationContext(DistributedApplicationModel appModel, IResource resource, Guid tenantId, Guid subscriptionId, string environmentName) : Infrastructure(ConstructScope.Subscription, tenantId, subscriptionId, environmentName)
```

Comment:
The bicep generation context provides access to information that will be useful during the Bicep generation. At the moment this is also an Azure.Provisioning IConstruct but this will likely change once we get a better feeling about how this will integrate.